core/src/main/scala/spinal/core/FixedPoint.scala

```diff
@@ -267,6 +279,18 @@ abstract class XFix[T <: XFix[T, R], R <: BitVector with Num[R]](val maxExp: Int
       case _ => SpinalError("Undefined assignment")
     }
   }
+
+  override def getMuxType[M <: Data](list: TraversableOnce[M]): HardType[M] =
+    HardType(fixFactory(maxExp, bitCount).asInstanceOf[M])
```

Comment:
I think that is wrong. It should provide the hardware type able to store every values from the list parameter.
For instance, if you have List(Q8.4, Q12.2) then it should return Q12.4

Reply:
And also, eventualy, if one of the input has the truncated flag, then that input width can be filtred out from the logic.

Reply:
Now that you say it - makes sense since XFix automatically upcasts - I'll change that.
Doesn't AFix have a similar problem (for different physical widths)?